dev_setup.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+# cryptol setup script
+#
+# Supported distribution(s):
+#  * macOS 14 (AArch64)
+#
+# This script installs everything needed to get a working development
+# environment for cryptol. Any new environment requirements should be
+# added to this script.
+#
+
+set -e
+
+HERE=$(cd `dirname $0`; pwd)
+LOG=$HERE/dev_setup.log
+
+function notice {
+    echo "[NOTICE] $*"
+}
+
+# Requires: LOG set to log file path.
+function logged {
+    if ! [ -z "$LOG" ]
+    then
+        mkdir -p `dirname $LOG`
+        echo $* >>$LOG
+        if ! $* >>$LOG 2>&1
+        then
+            echo
+            echo "An error occurred; please see $LOG"
+            echo "Here are the last 50 lines:"
+            tail -n 50 $LOG
+            exit 1
+        fi
+    fi
+}
+
+function update_submodules {
+    cd $HERE
```

Comment:
When I write these scripts I use `$HERE` to mean "the place where this script is located." That is sometimes not the same as the repository root, but it is used as a repository root here. I recommend adding an additional variable at the top, adjacent to the declaration of `HERE`, as `ROOT=$HERE` and then using `$ROOT` here. That way, if this script gets `git mv`'d somewhere else, only one line needs to be changed (the value of `ROOT`, relative to `HERE`).

Reply:
Thanks for clarifying, I fixed to use this convention.